framework/test/gtest/ServiceReferenceTest.cpp

```diff
@@ -88,8 +89,27 @@ class ServiceReferenceTest : public ::testing::Test
         framework.WaitForStop(std::chrono::milliseconds::zero());
     }
     Framework framework;
+    BundleContext context;
 };
 
+
+TEST_F(ServiceReferenceTest, TestServiceReferenceAssignmentAndCopy)
+{
+    auto s1 = std::make_shared<TestServiceA>();
+
+    ServiceRegistrationBase reg1 = context.RegisterService<ServiceNS::ITestServiceA>(s1);
+    ServiceReferenceBase ref1 = reg1.GetReference();
+
+    ASSERT_EQ(ref1.GetInterfaceId(), "");
+    ServiceReference<ServiceNS::ITestServiceA> typedRefAssign;
+    ServiceReference<ServiceNS::ITestServiceA> typedRefCopy = ref1;
```

Comment:
Perhaps:
```suggestion
    ServiceReference<ServiceNS::ITestServiceA> typedRefCopy{ref1};
```
to guarantee the copy constructor is always called, instead of copy assignment to a default constructed object?

Reply:
we created changes here to make sure the copy-assignment was doing the correct thing, which is what this test is trying to verify. 

If we did the constructor then we are just testing whether the already functional behavior worked right?

@Burgch